Update resolver

I am finding myself requiring newer Haskell packages and having to add them as `extra-deps`. It's a bit painful when those in turn have transitive dependencies that also need to be listed. It would be good to update the resolver.

I tried (to 13.xx and 14.{4,6}), but that failed with:
```
differential-datalog/src/Language/DifferentialDatalog/Compile.hs:1547:5: error:
    • Could not deduce (Control.Monad.Fail.MonadFail
                          Data.Functor.Identity.Identity)
        arising from a do statement
        with the failable pattern ‘Just aid’
      from the context: ?cfg::CompilerConfig
        bound by the type signature for:
                   mkAntijoin :: (?cfg::CompilerConfig) =>
                                 DatalogProgram
                                 -> [Int] -> Bool -> Atom -> Rule -> Int -> CompilerMonad Doc
        at src/Language/DifferentialDatalog/Compile.hs:1540:1-115
    • In a stmt of a 'do' block:
        Just aid <- getAntijoinArrangement atomRelation arr
      In the expression:
        do let ctx = CtxRuleRAtom rl ajoin_idx
           let (arr, _) = normalizeArrangement d ctx atomVal
           ffun <- mkFFun d rl input_filters
           Just aid <- getAntijoinArrangement atomRelation arr
           ....
      In an equation for ‘mkAntijoin’:
          mkAntijoin
            d
            input_filters
            input_val
            Atom {..}
            rl@Rule {..}
            ajoin_idx
            = do let ctx = ...
                 let (arr, _) = ...
                 ffun <- mkFFun d rl input_filters
                 ....
     |
1547 |     Just aid <- getAntijoinArrangement atomRelation arr
     |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```
(not sure if more errors await)

Haven't looked into it, instead filing this issue to get this done eventually.

--- a/package.yaml
+++ b/package.yaml
@@ -35,7 +35,7 @@ dependencies:
 - wide-word
 - graphviz
 - text
-- githash >= 0.1.3.3
+- githash
 - process
 - casing
 
@@ -98,7 +98,7 @@ tests:
     - -Werror
     dependencies:
     - differential-datalog
-    - tasty >= 1.2
+    - tasty
     - tasty-hunit
     - tasty-golden
     - filepath

--- a/src/Language/DifferentialDatalog/Compile.hs
+++ b/src/Language/DifferentialDatalog/Compile.hs
@@ -1547,7 +1547,7 @@ mkAntijoin d input_filters input_val Atom{..} rl@Rule{..} ajoin_idx = do
     let (arr, _) = normalizeArrangement d ctx atomVal
     -- Filter inputs using 'input_filters'
     ffun <- mkFFun d rl input_filters
-    Just aid <- getAntijoinArrangement atomRelation arr
+    aid <- fromJust <$> getAntijoinArrangement atomRelation arr
     next <- compileRule d rl ajoin_idx input_val
     return $ "XFormArrangement::Antijoin {"                                                                   $$
              "    description:" <+> (pp $ show $ show $ rulePPPrefix rl $ ajoin_idx + 1) <> ".to_string(),"   $$

--- a/stack.yaml
+++ b/stack.yaml
@@ -15,7 +15,7 @@
 # resolver:
 #  name: custom-snapshot
 #  location: "./custom-snapshot.yaml"
-resolver: lts-12.26
+resolver: lts-14.7
 
 # User packages to be built.
 # Various formats can be used as shown in the example below.
@@ -40,13 +40,6 @@ packages:
 # Dependency packages to be pulled from upstream that are not in the resolver
 # (e.g., acme-missiles-0.3)
 # extra-deps: []
-
-extra-deps:
-- ansi-terminal-0.10
-- ansi-wl-pprint-0.6.9
-- githash-0.1.3.3
-- graphviz-2999.20.0.3
-- tasty-1.2.3
 
 # Override default flag values for local packages and extra-deps
 # flags: {}

